### PR TITLE
Create release on new tags

### PR DESCRIPTION
**Pull request by rparkj12**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -42,11 +42,46 @@ jobs:
...
         # Artifact name
         name: build-release
         # A file, directory or wildcard pattern that describes what to upload
-        path: out/release/build-release.zip
\ No newline at end of file
+        path: out/release/build-release.zip
+
+    - name: Extract package version from cargo.toml
+      # set PACKAGE_VERSION from toml package
+      run: echo "::set-env name=PACKAGE_VERSION::$(cat proxy_agent/Cargo.toml | jq -r '.package.version')"
```
Comment (ZhidongPeng) [inline .github/workflows/ci.yml]: - name: version
    run: echo "::**set-output** name=**version**::$(cat proxy_agent/Cargo.toml | jq -r '.package.version')"
    id: **version**

${{ steps.version.outputs.**version** }}

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -42,11 +42,46 @@ jobs:
...
+      
+    - name: Create GitHub Release
+      id: create_release
+      uses: actions/create-release@v1
```
Comment (ZhidongPeng) [inline .github/workflows/ci.yml]: please create another workflow yml file to create a release when a tag is pushed

name: Releases

on: 
  push:
    tags:
    - '*'

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -42,11 +42,46 @@ jobs:
...
+        # ID of the release to attach the asset to
+        release_id: ${{ steps.create_release.outputs.id }}
+
+    - name: Upload release build to release
+      uses: actions/upload-release-asset@v1
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        #Path to the asset
+        asset_path: out/release/build-release.zip
+        asset_name: build-release.zip
```
Comment (ZhidongPeng) [inline .github/workflows/ci.yml]: please help add version into the asset_name

File: .github/workflows/cd.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+jobs:
+  release:
+
+    runs-on: windows-latest
+
+    steps:
+    - name: Get version from Cargo.toml
+      run: |
+        cargo install toml-cli
+        VERSION=$(toml get Cargo.toml package.version)
```
Comment (ZhidongPeng) [inline .github/workflows/cd.yml:52]: relative path? we have proxy_agent folder

File: .github/workflows/cd.yml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        echo "::set-output name=version::$VERSION"
+      id: version
+      
+    - name: Create GitHub Release
+      id: create_release
+      uses: actions/create-release@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        tag_name: ${{ github.ref }}
```
Comment (ZhidongPeng) [inline .github/workflows/cd.yml:62]: do you still need this tag_name here?

Comment (rparkj12) [reply .github/workflows/cd.yml:62]: yes, I think this is still needed, to set tag_ name with the tag reference

File: .github/workflows/cd.yml (hunk 4 above)
Comment (rparkj12) [reply .github/workflows/cd.yml:52]: fixed to relative path

**Review by ZhidongPeng — APPROVED** [review]

**Review by ZhidongPeng — APPROVED** [review]

**Pull request merged**